### PR TITLE
feat: Add support for pidfile

### PR DESCRIPTION
**Pull request by Shubhranshu153**
Issue #, if available:

*Description of changes:*
Add support for pidfile.

*Testing done:*
1. Start finch daemon with pidfile option and check if file is created.

- [ ] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/finch-daemon/main.go [hunk 1]
```diff
@@ -69,12 +74,34 @@ func runAdapter(cmd *cobra.Command, _ []string) error {
 	return run(options)
 }
 
+func handlePidFileOptions(options *DaemonOptions) error {
+	if options.PidFile != "" {
+		if err := os.MkdirAll(filepath.Dir(options.PidFile), 0o644); err != nil {
+			return fmt.Errorf("failed to create pidfile directory %s", err)
+		}
+		if err := pidfile.Write(options.PidFile, os.Getpid()); err != nil {
+			return fmt.Errorf("failed to start daemon, ensure finch daemon is not running or delete %s %s", options.PidFile, err)
```
Comment (pendo324) [inline cmd/finch-daemon/main.go:83]:
```suggestion
			return fmt.Errorf("failed to start daemon, ensure finch daemon is not running or delete %s %w", options.PidFile, err)
```

Comment (swagatbora90) [inline cmd/finch-daemon/main.go:83]: Error should be about failing to write pidfile

File: cmd/finch-daemon/main.go [hunk 2]
```diff
@@ -59,6 +63,7 @@ func main() {
 	rootCmd.Flags().IntVar(&options.socketOwner, "socket-owner", -1, "Uid and Gid of the server socket")
 	rootCmd.Flags().StringVar(&options.debugAddress, "debug-addr", "", "")
 	rootCmd.Flags().StringVar(&options.configPath, "config-file", defaultConfigPath, "Daemon Config Path")
+	rootCmd.Flags().StringVar(&options.PidFile, "pidfile", defaultPidFile, "Pid file location")
```
Comment (pendo324) [inline cmd/finch-daemon/main.go:66]: nit
```suggestion
	rootCmd.Flags().StringVar(&options.PidFile, "pidfile", defaultPidFile, "pid file location")
```

**Review by pendo324 — COMMENTED** [review]
LGTM besides a couple nits

File: cmd/finch-daemon/main.go [hunk 3]
```diff
@@ -42,6 +45,7 @@ type DaemonOptions struct {
 	socketOwner  int
 	debugAddress string
 	configPath   string
+	PidFile      string
```
Comment (swagatbora90) [inline cmd/finch-daemon/main.go:48]: nit: `pidFile`?

File: cmd/finch-daemon/main.go (hunk 1 above)
Comment (Shubhranshu153) [reply cmd/finch-daemon/main.go:83]: this is from docker convention

**Review by swagatbora90 — APPROVED** [review]

**Review by pendo324 — APPROVED** [review]

**Review by swagatbora90 — APPROVED** [review]

**Pull request merged**